### ISSUE TITLE
Improve:

### ISSUE THREAD

**Issue by Ommdutta26** [ronisarkarexe/ai-stories-book#3032]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### What would you like to improve?

# Improve Training Pipeline Reproducibility and Remove Duplicate Scaler Persistence

## Describe the feature

The training pipeline currently saves the fitted scaler multiple times and does not provide reproducible training runs.

#### 1. Duplicate Scaler Persistence

The scaler is saved inside `scale()` when `fit=True`:

```python
joblib.dump(scaler, SCALER_PATH)
```

and then saved again later in `train()`:

```python
joblib.dump(scaler, SCALER_PATH)
```

This duplicates responsibility for artifact persistence and makes the data preprocessing function perform file-system side effects.

#### 2. Non-Reproducible Training

The simulated training data relies on random generation:

```python
np.random.randint(...)
```

but no random seed is configured.

As a result:

* Training results vary between runs.
* Generated thresholds may differ.
* Hyperparameter tuning outcomes are harder to reproduce.
* Debugging model behavior becomes more difficult.

### Scaler Persistence

* Move scaler saving responsibility to a single location.
* Prefer keeping persistence logic in `train()` and keeping `scale()` focused on preprocessing.

### Reproducibility

* Introduce a configurable random seed (e.g. `RANDOM_SEED = 42`).
* Initialize NumPy's random generator before dataset generation.

## Benefits

* Cleaner separation of concerns.
* Easier maintenance.
* Reproducible experiments and model artifacts.
* More predictable threshold generation.
* Improved debugging and testing experience.

## Additional Context

This change is focused on training pipeline reliability and code quality. No model architecture changes are required.

### Record

- [x] I agree to follow this project's Code of Conduct
- [x] I want to work on this issue
